profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/DataCenterApiLiveTest.java

```diff
@@ -0,0 +1,112 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.profitbricks.rest.features;
+
+import java.util.List;
+import org.apache.jclouds.profitbricks.rest.domain.DataCenter;
+import org.apache.jclouds.profitbricks.rest.domain.Location;
+import org.apache.jclouds.profitbricks.rest.domain.options.DepthOptions;
+import org.apache.jclouds.profitbricks.rest.internal.BaseProfitBricksLiveTest;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import org.testng.annotations.Test;
+
+@Test(groups = "live", testName = "DataCenterApiLiveTest")
+public class DataCenterApiLiveTest extends BaseProfitBricksLiveTest {
+   
+   @Test
+   public void testGetList() {
+      List<DataCenter> list = dataCenterApi().list(new DepthOptions().depth(1));
+      assertNotNull(list);
+   }
+   
+   @Test
+   public void testCreate() {
+      DataCenter dataCenter = createDataCenter();
+      
+      assertNotNull(dataCenter);
+      assertNotNull(dataCenter.id());
+      
+      deleteDataCenter(dataCenter.id());
+   }
+   
+   @Test
+   public void testGetDataCenter() {
+      DataCenter dataCenter = createDataCenter();
+      
+      dataCenter = dataCenterApi().getDataCenter(dataCenter.id());
+      
+      assertNotNull(dataCenter);
+      assertEquals(dataCenter.properties().name(), "test-data-center");
+      
+      deleteDataCenter(dataCenter.id());
+   }
+      
+   @Test
+   public void testUpdate() {      
+      DataCenter dataCenter = createDataCenter();
+      
+      dataCenter = dataCenterApi().update(dataCenter.id(), "test-data-center2");
+      
+      assertNotNull(dataCenter);
+      assertEquals(dataCenter.properties().name(), "test-data-center2");
+      
+      deleteDataCenter(dataCenter.id());
+   }
+   
+   @Test
+   public void testDelete() {      
+      DataCenter dataCenter = createDataCenter();      
+      dataCenterApi().delete(dataCenter.id());      
+      dataCenter = getDataCenter(dataCenter.id());      
+      //assertEquals(dataCenter, null);
+   }
+   
+   @Test
+   public void testDepth() {
+      
+      DataCenter dataCenter = createDataCenter();
+      
+      for (int i = 1; i <= 5; ++i) {
+         dataCenter = getDataCenter(dataCenter.id(), new DepthOptions().depth(i));
+         assertNotNull(dataCenter);
```

Comment:
Sure, but I wouldn't go to far in, as we would need to have all the sub-items instantiated such as servers, lans.. and so on. (although we can come back to it, when these other parts of API are implemented)